### PR TITLE
Implement common methods for controllers to make sure basics are set

### PR DESCRIPTION
**Pull request by Quetzacoalt91**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Change the way some data are set on the controllers level, to trigger errors when a mandatory one is forgotten. This PR also enables the update options page.
| Type?             | improvement
| BC breaks?        | Nope
| Deprecations?     | Nope
| Fixed ticket?     | /
| Sponsor company   | /
| How to test?      | Loading the existing pages still works, and the page displaying the update options is reachable.

### CODE REVIEW

File: controllers/admin/self-managed/UpdatePagePostUpdateController.php
Comment (ga-devfront) [inline controllers/admin/self-managed/UpdatePagePostUpdateController.php]: Others steps page return `self::CURRENT_STEP` I think you have to do the same her.

**Review by ga-devfront — CHANGES_REQUESTED** [review]

File: controllers/admin/self-managed/UpdatePagePostUpdateController.php
Comment (Quetzacoalt91) [reply controllers/admin/self-managed/UpdatePagePostUpdateController.php]: You're right, this has been updated.

**Review by ga-devfront — APPROVED** [review]

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**
